applications/asset_tracker/CMakeLists.txt

```diff
@@ -22,9 +22,37 @@ zephyr_library_include_directories(
   src/orientation_detector
   src/ui
   src/cloud_codec
+  src/battery_monitor
   )
 
 # Application sources
 add_subdirectory(src/orientation_detector)
 add_subdirectory(src/ui)
 add_subdirectory(src/cloud_codec)
+add_subdirectory(src/gps_controller)
+if (CONFIG_BATTERY_MONITOR)
+add_subdirectory(src/battery_monitor)
+endif()
+
+include_directories(src/env_sensors)
+
+if (CONFIG_USE_BME680_BSEC)
+	target_sources(app PRIVATE src/env_sensors//bsec.c)
+
+	set (bsec_dir ${CMAKE_CURRENT_SOURCE_DIR}/src/env_sensors/BSEC_1.4.7.3_Generic_Release)
+	include_directories(${bsec_dir}/API)
+	target_sources(app PRIVATE ${bsec_dir}/examples/bsec_integration.c)
+	include_directories(${bsec_dir}/examples)
+	target_sources(app PRIVATE ${bsec_dir}/API/bme680.c)
+
+	set(BSEC_LIB_DIR ${bsec_dir}/algo/bin/Normal_version/gcc/Cortex_M4F)
+	set(BSEC_INCLUDE_DIR ${bsec_dir}/algo/bin/Normal_version/gcc/Cortex_M4F)
+	add_library(bsec_lib STATIC IMPORTED GLOBAL)
+	add_dependencies(bsec_lib math_lib bsec_target)
+	set_target_properties(bsec_lib PROPERTIES IMPORTED_LOCATION "${BSEC_LIB_DIR}/libalgobsec.a")
```

Comment:
You need to enforce somehow that the ABI of libalgobsec.a is compatible with the current build configuration.

See https://github.com/NordicPlayground/nrfxlib/blob/master/ble_controller/CMakeLists.txt#L14 for the current best-practice on how to do this.